src/toil_scripts/batch_alignment/bwa_alignment.py

```diff
@@ -308,19 +339,54 @@ def run_bwa(job, job_vars):
     work_dir = job.fileStore.getLocalTempDir()
     sudo = input_args['sudo']
     cores = input_args['cpu_count']
+
     # Retrieve samples
     return_input_paths(job, work_dir, ids, 'r1.fq.gz', 'r2.fq.gz')
+
     # Retrieve input files
     return_input_paths(job, work_dir, ids, 'ref.fa', 'ref.fa.amb', 'ref.fa.ann',
                        'ref.fa.bwt', 'ref.fa.pac', 'ref.fa.sa', 'ref.fa.fai')
+
+    # if alt file is provided, get the alts and modify the output name
+    if input_args['ref.fa.alt'] != None:
+        return_input_paths(job, work_dir, ids, 'ref.fa.alt')
+        output = 'aligned.alt.sam'
+    else:
+        output = 'aligned.sam'
+
     # Call: BWA
     parameters = ["mem",
                   "-t", str(cores),
                   "/data/ref.fa"] + ['/data/r1.fq.gz', '/data/r2.fq.gz']
 
-    with open(os.path.join(work_dir, 'aligned.sam'), 'w') as samfile:
-        docker_call(tool='quay.io/ucsc_cgl/bwa:0.7.12--dd5ac549b95eb3e5d166a5e310417ef13651994e',
+    with open(os.path.join(work_dir, output), 'w') as samfile:
+        docker_call(tool='quay.io/ucsc_cgl/bwa:0.7.12',
                     tool_parameters=parameters, work_dir=work_dir, outfile=samfile, sudo=sudo)
+    return job.fileStore.writeGlobalFile(os.path.join(work_dir, output))
+
+
+def lift_alts(job, job_vars, sam_id):
+    """
+    This lifts alt haplotype alignments back to the reference contigs that the
+    alts vary from. This is necessary for alt aware alignment against GRCh38.
```

Comment:
Could you clarify, is this necessary because bwa is being called without the ```-a``` parameter?  Are alignments copied over?  Many alignments might be dropped since whole genes (e.g. HLA-DRB5 on chr 6) and even larger regions (A vs B haplotypes in KIR region on chr 19) are not present in the reference chromosomes.